packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/ModelOperationCopy.java

```diff
@@ -0,0 +1,16 @@
+package software.uncharted.terarium.hmiserver.models.dataservice;
+
+import lombok.Data;
+import lombok.experimental.Accessors;
+
+import java.io.Serializable;
+
+@Data
+@Accessors(chain = true)
+public class ModelOperationCopy implements Serializable {
+	private Long left;
```

Comment:
Why the `left`?

Reply:
TDS uses a somewhat RDF-like schema for tracking, that being left/right/operation. RDF is subject/object/relation.

```
askem=# \d+ provenance
                                              
    Column     |            Type             | 
---------------+-----------------------------+-
 id            | integer                     | 
 timestamp     | timestamp without time zone | 
 relation_type | relationtype                | 
 left          | integer                     | 
 left_type     | provenancetype              | 
 right         | integer                     | 
 right_type    | provenancetype              | 
 user_id       | integer                     | 
 concept       | character varying           | 

```

Reply:
I agree, but why does the `hmi-client` needs to provide this when `id` is part of the URI already?

Reply:
This is a proxy object, the `/id` path param doesn't exist on TDS.

The API point we are hitting is
```
/models/opts/copy
```